### PR TITLE
Implements four new vector horizontal ALU ops 

### PR DESCRIPTION
**Pull request by Sonicadvance1**
haddps, hsubps, haddpd, hsubpd

### CODE REVIEW

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp [hunk 1]
```diff
@@ -5303,6 +5303,46 @@ void OpDispatchBuilder::MOVBEOp(OpcodeArgs) {
...
+  StoreResult(FPRClass, Op, Res, -1);
+}
+
+template<size_t ElementSize>
+void OpDispatchBuilder::HSUBP(OpcodeArgs) {
+  auto Size = GetSrcSize(Op);
+  OrderedNode *Dest = LoadSource(FPRClass, Op, Op->Dest, Op->Flags, -1);
+  OrderedNode *Src = LoadSource(FPRClass, Op, Op->Src[0], Op->Flags, -1);
+
+  // This is a bit complicated since AArch64 doesn't support a pairwise subtract
```
Comment (phire) [inline External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:5322]: I hope this leads to some good optimisation opportunities.

File: External/FEXCore/Source/Interface/Core/Interpreter/InterpreterCore.cpp [hunk 2]
```diff
@@ -2371,6 +2386,23 @@ void InterpreterCore::ExecuteCode(FEXCore::Core::InternalThreadState *Thread) {
             memcpy(GDP, Tmp, OpSize);
             break;
           }
+          case IR::OP_VFADDP: {
+            auto Op = IROp->C<IR::IROp_VFAddP>();
+            void *Src1 = GetSrc<void*>(Op->Header.Args[0]);
+            void *Src2 = GetSrc<void*>(Op->Header.Args[1]);
+            uint8_t Tmp[16];
+
+            uint8_t Elements = OpSize / Op->Header.ElementSize / 2;
```
Comment (phire) [inline External/FEXCore/Source/Interface/Core/Interpreter/InterpreterCore.cpp:2395]: Could we have some brackets so I don't have to remember operator precedence rules?

File: External/FEXCore/Source/Interface/Core/Interpreter/InterpreterCore.cpp [hunk 3]
```diff
@@ -2394,7 +2426,7 @@ void InterpreterCore::ExecuteCode(FEXCore::Core::InternalThreadState *Thread) {
             void *Src2 = GetSrc<void*>(Op->Header.Args[1]);
             uint8_t Tmp[16];
 
-            uint8_t Elements = OpSize / Op->Header.ElementSize;
+            uint8_t Elements = OpSize / Op->Header.ElementSize / 2;
```
Comment (phire) [inline External/FEXCore/Source/Interface/Core/Interpreter/InterpreterCore.cpp:2429]: Same here

Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/Interpreter/InterpreterCore.cpp]: Concern addressed

File: External/FEXCore/Source/Interface/Core/Interpreter/InterpreterCore.cpp (hunk 2 above)
Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/Interpreter/InterpreterCore.cpp]: Concern addressed

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp (hunk 1 above)
Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:5322]: I hope so as well :)

**Pull request merged**